soco/zgs.py

```diff
@@ -0,0 +1,267 @@
+"""Provides handling for ZoneGroupState information.
+
+ZoneGroupState XML payloads are received from both:
+* zoneGroupTopology.GetZoneGroupState()['ZoneGroupState']
+* zoneGroupTopology subscription event callbacks
+
+The ZoneGroupState payloads are identical between all speakers in a
+household, but may be generated with differing orders for contained
+ZoneGroup or ZoneGroupMember elements and children. To benefit from
+similar contents, payloads are passed through an XSL transformation
+to normalize the data, to allow simple equality comparisons, and to
+avoid unnecessary reprocessing of identical data.
+
+Since the payloads are identical between all speakers, we can use a
+common cache per household.
+
+Satellite devices can sometimes deliver outdated payloads when they
+are directly polled. These requests are instead redirected to their
+parent device.
```

Comment:
```suggestion
As satellite devices can sometimes deliver outdated payloads when they
are directly polled, these requests are instead redirected to their
parent device.
```
If I'm reading correctly, that only the parent is being accessed no matter whether "these" requests are made to any of the children?